### DISCUSSION TITLE
How to mock RNS with Jest

### DISCUSSION THREAD

**Discussion by getinnocuous** [software-mansion/react-native-screens#838] [Q&A] [answered]
Hi

Has anyone come up with a way to mock RNS? 

Jest is blowing up with the following error

`Native stack is only available if React Native Screens is enabled.`

**Comment by kacperkapusciak** [comment, marked as answer]
Hey @getinnocuous,
try this snippet:
```Javascript
jest.useFakeTimers();
jest.mock('react-native-screens', () => {
  const RealComponent = jest.requireActual('react-native-screens');
  RealComponent.enableScreens = function () {};
  RealComponent.screensEnabled = function () {
    return true;
  };
  return RealComponent;
});
```

If you want to know more about this see: https://github.com/software-mansion/react-native-screens/issues/247#issuecomment-569663511 and this: https://github.com/facebook/jest/issues/6434#issuecomment-525576660

If this snippet doesn't help please provide us with some code to reproduce your problem. Cheers!

**Reply by getinnocuous** [reply]
This works - thanks so much!
